src/moov/trak/mdia/minf/stbl/stsd/audio.rs

```diff
@@ -0,0 +1,66 @@
+use crate::coding::{Decode, Encode};
+use crate::{Buf, BufMut, Error, FixedPoint, Result};
+
+#[cfg_attr(feature = "serde", derive(serde::Serialize, serde::Deserialize))]
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct Audio {
+    pub data_reference_index: u16,
+    pub channel_count: u16,
+    pub sample_size: u16,
+    pub sample_rate: FixedPoint<u16>,
+}
+
+impl Encode for Audio {
+    fn encode<B: BufMut>(&self, buf: &mut B) -> Result<()> {
+        0u32.encode(buf)?; // reserved
+        0u16.encode(buf)?; // reserved
+        self.data_reference_index.encode(buf)?;
+        0u16.encode(buf)?; // version
+        0u16.encode(buf)?; // reserved
+        0u32.encode(buf)?; // reserved
+        self.channel_count.encode(buf)?;
+        self.sample_size.encode(buf)?;
+        0u32.encode(buf)?; // reserved
+        self.sample_rate.encode(buf)?;
+
+        Ok(())
+    }
+}
+impl Decode for Audio {
+    fn decode<B: Buf>(buf: &mut B) -> Result<Self> {
+        u32::decode(buf)?; // reserved
+        u16::decode(buf)?; // reserved
+        let data_reference_index = u16::decode(buf)?;
+        let version = u16::decode(buf)?;
+        u16::decode(buf)?; // reserved
+        u32::decode(buf)?; // reserved
+        let channel_count = u16::decode(buf)?;
+        let sample_size = u16::decode(buf)?;
+        u32::decode(buf)?; // pre-defined, reserved
+        let sample_rate = FixedPoint::decode(buf)?;
+
+        match version {
+            0 => {}
```

Comment:
It looks like there can be extra boxes here (ISO/IEC 14496-12:2022 Section 12.2.3.2), which could be stashed away as an `vec` of `Unknown`. I'm planning something similar for `Visual`, and can add that here as well if that would be useful.